src/main/scala/graphql/schema/SpatialProfileType.scala

```diff
@@ -75,20 +75,22 @@ object SpatialProfileType {
           case Type.Point.name    => SpatialProfile.PointSource.asRight
           case Type.Uniform.name  => SpatialProfile.UniformSource.asRight
           case Type.Gaussian.name =>
-            c.downField("fwhm").as[Double].map { fwhm => SpatialProfile.GaussianSource(fwhm) }
+            c.downField("fwhm").as[Double].map { fwhm =>
+              SpatialProfile.GaussianSource.arcsec.reverseGet(fwhm)
+            }
         }
         // Should never fail due to the way the schema is defined.
         r.getOrElse(throw new IllegalArgumentException(s"Can't decode ${json.noSpaces} as a SpatialProfile."))
       }
 
     implicit val ToInputSpatialProfileJson: ToInput[SpatialProfile, Json] =
       ToInput[Json, Json].contramap {
-        case SpatialProfile.PointSource          => Json.obj("type" -> Type.Point.name.asJson)
-        case SpatialProfile.UniformSource        => Json.obj("type" -> Type.Uniform.name.asJson)
-        case SpatialProfile.GaussianSource(fwhm) =>
+        case SpatialProfile.PointSource           => Json.obj("type" -> Type.Point.name.asJson)
```

Comment:
Can we move the `Json.obj(...)` to the outside? any benefits on doing so?

Reply:
:+1: That makes sense to me, to factor out the common part.

Reply:
Hmm, actually now that I've taken a closer look I think it makes the code a little worse.  It would become something like

```scala
implicit val ToInputSpatialProfileJson: ToInput[SpatialProfile, Json] =
  ToInput[Json, Json].contramap { sp =>
    Json.fromFields(
      sp match {
            case SpatialProfile.PointSource           => List("type" -> Type.Point.name.asJson)
            case SpatialProfile.UniformSource         => List("type" -> Type.Uniform.name.asJson)
            case g @ SpatialProfile.GaussianSource(_) =>
              List(
                "type" -> Type.Gaussian.name.asJson,
                "fwhm" -> SpatialProfile.GaussianSource.arcsec.get(g).asJson
              )
      }    
    )        
```
which doesn't seem better.  Or maybe I'm missing the obvious?

Reply:
Yeah maybe it is not any better